app/helpers/article_helper.rb

```diff
@@ -15,10 +15,14 @@ def link_subjects(options = {})
     return unless options[:value]
     separators = options.dig(:config, :separator_options) || {}
     values = render_text_from_html(options[:value])
-    values.collect do |value|
+    if params[:action] == 'index'
```

Comment:
I think you can implement this within ArticleController by omitting the helper method `link_subjects` for the index pages. If you look in there, check out how we ignore the `separator_options` for index pages -- see https://github.com/sul-dlss/SearchWorks/blob/article-search/app/controllers/article_controller.rb#L121